OfflineAudioContext/explainer.md

```diff
@@ -0,0 +1,203 @@
+
+Memory usage is bounded by the size of each chunk plus the backlog of unhandled buffers.
+
+### Questions
+
+- What should return of `ReadableStreamDefaultReader.read()` be? Float32Array with BYOB or AudioBuffer?
+
+### Pros
+
+- Aligns well with other web streaming APIs, similar to [WebCodecs](https://streams.spec.whatwg.org/#readablestream)
+- Works with very large durations, no upper limit to WebAudio graph duration
+- Flexible usage scenarios for the consumers
+
+### Cons
+
+- Requires spec change
+- Need to define sensible chunk sizes, backpressure, error handling, and end-of-stream
+
+### Implement OfflineAudioContext.startRendering() streaming behaviour with this approach
+
+#### Option 1: AudioBuffer stream
+```js
+const offlineContext = new OfflineAudioContext(...);
+
+// ... build up WebAudio graph
+
+const stream = await offlineContext.startRendering(options: { mode: "stream"});
+const reader = stream.getReader();
+while (true) {
+    // get the next chunk of data from the stream
+    const result = await reader.read();
+
+    // the reader returns done = true when there are no more chunks to consume
+    if (result.done) {
+        break;
+    }
+
+    const buffers = result.value;
+}
+```
+
+#### Option 2: BYOB reading with Float32Array stream
+```js
+/**
+ * New API
+ */
+const offlineContext = new OfflineAudioContext(...);
+
+// ... build up WebAudio graph
+
+const stream = await offlineContext.startRendering(options: { mode: "stream"});
+const reader = stream.getReader({ mode: 'byob' });
+let buffer = new ArrayBuffer(...);
+while (true) {
+    const result = await reader.read(new Float32Array(buffer));
+
+    // the reader returns done = true when there are no more chunks to consume
+    if (result.done) {
+        break;
+    }
+
+    // process result...
+
+    buffer = result.value.buffer;
+}
+```
+
+In both cases, the existing API remains unchanged for backwards compatability:
+```js
+/**
+ * Existing API unchanged
+ */
+const offlineContext = new OfflineAudioContext(...);
+
+// ... build up WebAudio graph
+
+// Full AudioBuffer is allocated
+const renderedBuffer = await offlineContext.startRendering();
+```
+
+## Alternatives considered
+
+### [Alternative 1]
+
+Keep current `startRendering()` API but do not allocate the full `AudioBuffer`. After starting, periodically emit events on the context or a new interface such as `ondataavailable(chunk: AudioBuffer)`.
+
+The user can subscribe and collect chunks for processing.
+
+At the end, the API may optionally still provide a full `AudioBuffer`.
+
+#### Pros
+
+-   Simple to integrate with existing event-driven patterns.
+
+#### Cons
+
+
+-   Chunking semantics need spec
+-   Memory benefit only if user discards chunks but at least this is the in user control
+
+#### Concerns
+
+-   Browser vendors may implement the chunking API but still allocate full buffer internally defeating memory reduction goal unless spec mandates avoiding full allocation.
```

Comment:
This is not really a concern in my opinion, because we are already stating above that "Keep current `startRendering()` API but _do not_ allocate the full `AudioBuffer`." So this would be written down in the specification somewhere if we were to adopt this alternative.

I would suggest removing this "concerns" subsection.